### DISCUSSION TITLE
Support ThemeExtension

### DISCUSSION THREAD

**Discussion by fischerscode** [bdlukaa/fluent_ui#672] [Ideas]
Material supports [ThemeExtensions](https://medium.com/codex/flutter-skill-of-themeextensions-and-material3-3e3e95bf591d). I think it would be nice if fluent_ui would provide something similar. The [implementation](https://github.com/flutter/flutter/commit/8c1c2f6af5558c27946e8376541649090073d50c#diff-9d9b537da8ee2b1c3d29615749e838767109e585bd59cdbd4ac356e30f344505) seems to be not that difficult.

In particular, I'd like to use [theme_tailor](https://pub.dev/packages/theme_tailor) in my fluent_ui project.

Is something like this considered to be out of scope for fluent_ui or might a PR get accepted?

**Comment by bdlukaa** [comment]
Yeah, it'd totally be accepted! I was planning to add support for it, but I haven't got the time for it. The package already depends on a lot of material resources, so I believe adding `ThemeExtension` support is a great thing :)

**Comment by bdlukaa** [comment]
On the latest versions, you can use `FluentThemeData.extensions`, which has a similar API to material's `ThemeData.extension`
